### PR TITLE
Introduce property 'Has a closed point' (P107)

### PR DESCRIPTION
**Pull request by GeoffreySangston**
Discussed in Issue https://github.com/pi-base/data/issues/940

Work in progress.

### CODE REVIEW

**Comment by prabau** [comment]
Maybe, I have not read it yet.  But I think @StevenClontz was suggesting in #940 to not add all possible theorems about this notion, but rather only introduce the definition and a few obvious theorems, together with some simplification of existing theorems and some associated trait cleanup for just one space, to show the utility of the new concept.  More theorems and changes can then be added in a separate PR.

**Comment by GeoffreySangston** [comment]
@prabau Okay. We'll add it back later.

As a sign of future utility then, with it currently added the systems automatically derives 'Double pointed reals is not Has a cut point' [S54|P205](https://topology.pi-base.org/spaces/S000054/properties/P000204).

**Comment by GeoffreySangston** [comment]
S150 already had an optimized list of traits. I thought the argument for 'Has a closed point' was strictly simpler, but on second thought it's not that clear that 'Has a closed point' is a simpler property than 'Has a point with a unique neighborhood'. So I can look for a different demonstration.

**Comment by prabau** [comment]
I thought there were some existing theorems that can be modified/simplified if we use the notion of closed point.  I can't think of anything off hand right now.  But maybe later.

**Comment by GeoffreySangston** [comment]
> I thought there were some existing theorems that can be modified/simplified if we use the notion of closed point. I can't think of anything off hand right now. But maybe later.

This PR factors main's T618 into T618 and T632

**Comment by prabau** [comment]
I just realized #959 also introduces a property P205.  Any chance this one could be changed to P206, or maybe a previously unused number (there are quite a few)?

**Comment by GeoffreySangston** [comment]
[The topological sum of Sierpinski space and a singleton (S188)](https://topology.pi-base.org/spaces/S000188) could support a simple demonstration of this property's utility. [S188 asserts ~T1](https://topology.pi-base.org/spaces/S000188/properties/P000002) and has a closed point, so the [derivation of ~homogeneous](https://topology.pi-base.org/spaces/S000188/properties/P000086), which currently uses  could be simplified. I'm not actually sure if the system will produce with simpler derivation though.

(In post.. scratch that. After learning the definition of 'Almost discrete' from looking at this, I see that 'Almost discrete => Has a closed point' is another obvious theorem to add. So it won't be asserted as a trait here. Maybe the derivation of ~Homogeneous will shorten though.)

**Comment by GeoffreySangston** [comment]
I'm not immediately seeing anything super convincing without the deleted theorem, so I added 'Almost discrete => Has a close point', to hopefully make it easier to look around.

**Comment by prabau** [comment]
Well, I personally am convinced of the utility of this property.  You could go ahead of mark it ready for review, and we can go from there, and not approve without Steven's comments.

**Comment by david20000813** [comment]
I also think this seems fine. We'll wait for Steven to give his comments.

**Comment by prabau** [comment]
I added myself as a reviewer, to give me the opportunity to read the PR and comment in case it get approved by someone else in the mean time.

**Review by StevenClontz — COMMENTED** [review]
Looks good to me. Leaving it to @prabau to mark as approved or provide feedback since he's self-requested a review.

**Comment by prabau** [comment]
P107 has a useful link to P139 (Has an isolated point).  So we can also have a link in the other direction.

File: spaces/S000150/properties/P000107.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+space: S000150
+property: P000107
+value: true
+---
+
+The interval $(0, \rightarrow)$ is open, by definition, so the singleton $\{0\}$ is closed.
```
Comment (prabau) [inline spaces/S000150/properties/P000107.md:7]:
```suggestion
The interval $(0, \rightarrow)$ is open, so the singleton $\{0\}$ is closed.
```

**Comment by david20000813** [comment]
> P107 has a useful link to P139 (Has an isolated point). So we can also have a link in the other direction.

@prabau Added.

**Comment by StevenClontz** [comment]
Can/should we add in a theorem that will allow us to close #942 when this is merged?

**Comment by prabau** [comment]
Approving and merging to get things going.  Additional theorems can be added seperately.

**Review by prabau — APPROVED** [review]

**Comment by david20000813** [comment]
Should we add the originally proposed T631 now? That would close #942, if I’m not mistaken.

**Pull request merged**